app/v4/retrieveHistoricNonFhlUkPropertyAnnualSubmission/RetrieveHistoricNonFhlUkPropertyAnnualSubmissionConnector.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v4.retrieveHistoricNonFhlUkPropertyAnnualSubmission
+
+import api.connectors.DownstreamUri.IfsUri
+import api.connectors.httpparsers.StandardDownstreamHttpParser.reads
+import api.connectors.{BaseDownstreamConnector, DownstreamOutcome}
+import config.AppConfig
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import v4.retrieveHistoricNonFhlUkPropertyAnnualSubmission.model.request.{
+  Def1_RetrieveHistoricNonFhlUkPropertyAnnualSubmissionRequestData,
+  RetrieveHistoricNonFhlUkPropertyAnnualSubmissionRequestData
+}
+import v4.retrieveHistoricNonFhlUkPropertyAnnualSubmission.model.response.{
+  Def1_RetrieveHistoricNonFhlUkPropertyAnnualSubmissionResponse,
+  RetrieveHistoricNonFhlUkPropertyAnnualSubmissionResponse
+}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class RetrieveHistoricNonFhlUkPropertyAnnualSubmissionConnector @Inject() (val http: HttpClient, val appConfig: AppConfig)
+    extends BaseDownstreamConnector {
+
+  def retrieve(request: RetrieveHistoricNonFhlUkPropertyAnnualSubmissionRequestData)(implicit
+      hc: HeaderCarrier,
+      ec: ExecutionContext,
+      correlationId: String): Future[DownstreamOutcome[RetrieveHistoricNonFhlUkPropertyAnnualSubmissionResponse]] = {
+
+    request match {
+      case def1: Def1_RetrieveHistoricNonFhlUkPropertyAnnualSubmissionRequestData =>
+        import def1._
+
+        val downstreamUri =
+          IfsUri[Def1_RetrieveHistoricNonFhlUkPropertyAnnualSubmissionResponse](
+            s"income-tax/nino/$nino/uk-properties/other/annual-summaries/${taxYear.asDownstream}")
+
+        val response = get(downstreamUri)
```

Comment:
can shorten this to just get(downstreamUri)

Reply:
Annoyingly, it's needed to keep the Scala compiler happy